api-extractor/src/definitions/ApiNamespace.ts

```diff
@@ -0,0 +1,108 @@
+
+/**
+  * This class is part of the ApiItem abstract syntax tree. It represents exports of
+  * a namespace, the exports can be classes, interfaces, type literals, enums, functions,
+  * and properties.
+  */
+export default class ApiNamespace extends ApiItemContainer {
+  private _exportedNormalizedSymbols: IExportedSymbol[] = [];
+
+  constructor(options: IApiItemOptions) {
+    super(options);
+    this.kind = ApiItemKind.Namespace;
+    this.name = options.declarationSymbol.name;
+
+    const exportSymbols: ts.Symbol[] = this.typeChecker.getExportsOfModule(this.declarationSymbol);
+    if (exportSymbols) {
+      for (const exportSymbol of exportSymbols) {
+        const followedSymbol: ts.Symbol = this.followAliases(exportSymbol);
+
+        if (!followedSymbol.declarations) {
+          this.reportWarning(`Definition with no declarations: ${exportSymbol.name}`);
+          continue;
+        }
+
+        if (!(followedSymbol.flags === ts.SymbolFlags.BlockScopedVariable)) {
+          this.reportWarning(`Unsupported export "${exportSymbol.name}" ` +
+            'ApiNamespace only supports properties.');
+          continue;
+        }
+
+        // Since we are imposing that the items within a namespace be
+        // const properties we are only taking the first declaration.
+        // If we decide to add support for other types within a namespace
+        // we will have for evaluate each declaration.
+        const declaration: ts.Declaration = followedSymbol.getDeclarations()[0];
+
+        if (declaration.parent.flags !== ts.NodeFlags.Const) {
+          this.reportWarning(`Export "${exportSymbol.name}" must possess the "const" modifier`);
+          continue;
+        }
+
+        const propertySignature: ts.PropertySignature = declaration as ts.PropertySignature;
+        const type: string = propertySignature.type.getText();
+        if (allowedTypes.indexOf(type) < 0) {
+          this.reportWarning(`Export "${exportSymbol.name}" must of type "string", "number" or "boolean"`);
+          continue;
+        }
+
+        const exportMemberOptions: IApiItemOptions = {
+          extractor: this.extractor,
+          declaration,
+          declarationSymbol: followedSymbol,
+          jsdocNode: declaration,
+          exportSymbol
+        };
+
+        this.addMemberItem(new ApiField(exportMemberOptions));
+
+        this._exportedNormalizedSymbols.push({
+          exportedName: exportSymbol.name,
+          followedSymbol: followedSymbol
+        });
+      }
+    }
+  }
+
+  /**
+   * Finds and returns the original symbol name.
+   *
+   * For example, suppose a class is defined as "export default class MyClass { }"
+   * but exported from the package's index.ts like this:
+   *
+   *    export { default as _MyClass } from './MyClass';
+   *
+   * In this example, given the symbol for _MyClass, getExportedSymbolName() will return
+   * the string "MyClass".
+   */
+  public tryGetExportedSymbolName(symbol: ts.Symbol): string {
```

Comment:
>tryGetExportedSymbolName [](start = 9, length = 24)

Why do we need this?  I would expect ApiNamespace to be more like ApiStructuredType than ApiPackage.